tests/test_utils.py

```diff
@@ -519,23 +519,44 @@ def test_strip_registry_and_tag_from_image(img, expected):
 
 def test_checkout_git_repo(tmpdir):
     repo_path = tmpdir.mkdir("repo").strpath
-    initialize_git_repo(repo_path)
-    with checkout_git_repo(repo_path, 'HEAD') as repo:
+    initialize_git_repo(repo_path, ['file1'])
+    with checkout_git_repo(repo_path, 'HEAD') as repo_info:
+        repo = repo_info[0]
         assert os.path.isdir(repo)
         assert os.path.isdir(os.path.join(repo, '.git'))
         assert subprocess.check_output(['git', 'status'], cwd=repo)
 
+    deep_repo_path = tmpdir.mkdir("deep_repo").strpath
+    first_commit_ref = initialize_git_repo(deep_repo_path, ['file1', 'file2', 'file3', 'file4'])
+    with checkout_git_repo(deep_repo_path, first_commit_ref, 'master') as deep_repo_info:
+        deep_repo = deep_repo_info[0]
```

Comment:
I hoped for accessing attributes by name
```python
deep_repo = deep_repo_info.repo_path 
depth = deep_repo_info.commit_depth
```

Reply:
Fixed, thanks.